Upgrade ESLint config and use Silvermine's custom commitlint config

jimjenkins5 opened the pull request.

jimjenkins5 committed:
chore: upgrade @silvermine/eslint-confg to v2.1.0

jimjenkins5 committed:
chore: Switch to using Silvermine's custom commitlint config

jthomerson commented on package.json:
```diff
@@ -24,10 +24,9 @@
    },
    "devDependencies": {
       "@commitlint/cli": "7.5.2",
```
For posterity: ADIRL, we'll take this out since the travis-cli dependency has this as a dependency. That way upgrades are easier (the two are in sync).

jimjenkins5 committed:
chore: Remove @commitlint/cli from devDependencies

@commitlint/cli is a dependency of @commitline/travis-cli, so we don't
need to add it to our devDependencies to run commitlint locally.
Removing it from our devDependencies also ensures that when it is run
locally it will be run with the same version that travis will use.
